develop/optimize-sql-best-practices.md

```diff
@@ -3,3 +3,155 @@ title: Performance Tuning Best Practices
 ---
 
 # Performance Tuning Best Practices
+
+This section introduce some best practices for using TiDB databases.
+
+## DML Best Practices
+
+### Use multi-row statement
+
+When you need to modify multiple rows of table, it is recommended to use multi-row statement:
+
+```sql
+INSERT INTO t VALUES (1, 'a'), (2, 'b'), (3, 'c');
+
+DELETE FROM t WHERE id IN (1, 2, 3);
+```
+
+Not recommended to use multiple single-row statements:
+
+```sql
+INSERT INTO t VALUES (1, 'a');
+INSERT INTO t VALUES (2, 'b');
+INSERT INTO t VALUES (3, 'c');
+
+DELETE FROM t WHERE id = 1;
+DELETE FROM t WHERE id = 2;
+DELETE FROM t WHERE id = 3;
+```
+
+### Use PREPARE
+
+When you need to execute a SQL statement multiple times, it is recommended to use the `PREPARE` statement to avoid the overhead of repeatedly parse the SQL syntax.
+
+<SimpleTab>
+<div label="Golang">
+
+```go
+func BatchInsert(db *sql.DB) error {
+	stmt, err := db.Prepare("INSERT INTO t (id) VALUES (?), (?), (?), (?), (?)")
+	if err != nil {
+		return err
+	}
+	for i := 0; i < 1000; i += 5 {
+		values := []interface{}{i, i + 1, i + 2, i + 3, i + 4}
+		_, err = stmt.Exec(values...)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
```suggestion
```go
func BatchInsert(db *sql.DB) error {
    stmt, err := db.Prepare("INSERT INTO t (id) VALUES (?), (?), (?), (?), (?)")
    if err != nil {
        return err
    }
    for i := 0; i < 1000; i += 5 {
        values := []interface{}{i, i + 1, i + 2, i + 3, i + 4}
        _, err = stmt.Exec(values...)
        if err != nil {
            return err
        }
    }
    return nil
}
```